services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/builders/terraform/TFExecutor.java

```diff
@@ -0,0 +1,139 @@
+package org.eclipse.osc.orchestrator.plugin.huaweicloud.builders.terraform;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.FileSystems;
+import java.util.Arrays;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.osc.orchestrator.plugin.huaweicloud.exceptions.TFExecutorException;
+import org.eclipse.osc.services.ocl.loader.Ocl;
+
+@Slf4j
+public class TFExecutor {
+
+    private final Ocl ocl;
+    private final Map<String, String> env;
+    private String tfPath;
+    private String workPath;
+
+    public TFExecutor(final Ocl ocl, Map<String, String> env) {
+        this.ocl = ocl;
+        this.env = env;
+    }
+
+    public void createWorkspace() {
+        File ws =
+            new File("terraform_ws" + FileSystems.getDefault().getSeparator() + ocl.getName());
+
+        if (!ws.exists() && !ws.mkdirs()) {
+            throw new TFExecutorException(
+                "Create workspace for TFExecutor failed, File path not created: "
+                    + ws.getAbsolutePath());
+        }
+
+        tfPath = ws.getAbsolutePath() + FileSystems.getDefault().getSeparator() + "main.tf";
+        String verPath =
+            ws.getAbsolutePath() + FileSystems.getDefault().getSeparator() + "/version.tf";
+        workPath = ws.getAbsolutePath();
```

Comment:
It's ok for now, but suggest to use obs as backend storage to store tf state file in future.
https://registry.terraform.io/providers/huaweicloud/huaweicloud/latest/docs/guides/remote-state-backend

Reply:
Agree.
This is a temporary working directory, the valuable information needs to be analyzed and pushed to persistent storage. 
We can pull them out to a new working directory again for resource updating/destruction.

Reply:
ok, we can do more further discussion about this later.